src/results.js

```diff
@@ -6,11 +6,34 @@ function timeMs() {
   return new Date().getTime()
 }
 
-function Preview({url}) {
-  let embedUrl = `${url}?embed=on`
-  return (
-    <iframe className='results-item-preview' src={embedUrl} />
-  )
+class Preview extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {loaded: false}
+    this.loadHandler = this.loadHandler.bind(this)
+  }
+
+  loadHandler() {
+    this.setState({loaded: true})
+  }
+
+  render() {
+    let embedUrl = `${this.props.url}?embed=on`
+    let style = (!this.state.loaded && {display: 'none'}) || {}
```

Comment:
I think you can write this as `let style = {dislpay: this.state.loaded}` and react will compile it to the correct inline css.